plugin/pkg/admission/priority/admission.go

```diff
@@ -211,12 +213,17 @@ func (p *priorityPlugin) admitPod(a admission.Attributes) error {
 			}
 
 			priority = pc.Value
+
+			if utilfeature.DefaultFeatureGate.Enabled(features.NonPreemptingPriority) {
```

Comment:
make this unconditional, and put the feature gate around the `pod.Spec.PreemptionPolicy = preemptionPolicy` check below